presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkExecutionException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+public abstract class PrestoSparkExecutionException
+        extends RuntimeException
+{
+    protected PrestoSparkExecutionException(String message, String encodedExecutionFailureInfo, Throwable cause)
+    {
+        super(formatExceptionMessage(message, encodedExecutionFailureInfo), cause);
+    }
+
+    private static String formatExceptionMessage(String message, String encodedExecutionFailureInfo)
+    {
+        return message + " | ExecutionFailureInfo[" + encodedExecutionFailureInfo + "] |";
```

Comment:
So wrap the encoded execution info with "|" is to allow being parsed out in `PrestoSparkExecutionExceptionFactory` ?

Reply:
Also it helps to visually divide it when reading the error message from the executor